docs-src/concepts.md

```diff
@@ -343,6 +343,30 @@ status:
     Message: "foobar" is an FooBar.
 ```
 
+### GatewayClass controller selection
+
+The `GatewayClass.spec.controller` field is used to determine whether
+or not a given `GatewayClass` is managed by the controller.
+This format of this field is opaque and specific to a particular controller.
+Which GatewayClass is selected by a given controller field depends on how
+various controller(s) in the cluster interpret this field.
+
+It is RECOMMENDED that controller authors/deployments make their
+selection unique by using a domain / path combination under their
+administrative control (e.g. controller managing of all `controller`s
+starting with `acme.io` is the owner of the `acme.io` domain) to avoid
+conflicts.
+
+Controller versioning can be done by encoding the version of a
+controller into the path portion. An example scheme could be (similar
+to container URIs):
+
+```text
+acme.io/gateway/v1.14   // Use version 1.14
```

Comment:
Even the `.` is invalid according to the upstream Kubernetes validation. Each segment of the path must be an DNS 1123 label as defined here: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L177

Reply:
My suggestion would be to just use v1 and v2 for simplicity in this example. In practice you could also express this as `v1-14`, though that's not ideal.

Reply:
I read the following:

```
// IsDomainPrefixedPath checks if the given string is a domain-prefixed path
// (e.g. acme.io/foo). All characters before the first "/" must be a valid
// subdomain as defined by RFC 1123. All characters trailing the first "/" must
// be valid HTTP Path characters as defined by RFC 3986.
func IsDomainPrefixedPath(fldPath *field.Path, dpPath string) field.ErrorList {
```
and updated the doc as such.

We shouldn't do such a validation just for the sake of it and rather relax validation to include dot in our case, because that's what makes most sense for implementers and users.

Reply:
I don't mind the idea of more relaxed validation here in the future, I'd just prefer to intentionally discuss/choose that separately. For now I think it's safest to suggest a value that would be safe for both IngressClass and GatewayClass. For example `acme.io/gateway/v1` and `acme.io/gateway/v2` communicate the same use case and are fully valid for both APIs.

Reply:
I see. Updated for now. This is a minor detail but still would like to track it in #330.

Reply:
For the record, I was wrong here: https://github.com/kubernetes-sigs/gateway-api/issues/330#issuecomment-844588025.